### PR TITLE
Add asyncNotifyException method (for one-level deep async stacktraces)

### PR DESCRIPTION
**Pull request by jmshal**
See https://github.com/bugsnag/bugsnag-js/issues/241#issuecomment-305942433 for more context & example.

### CODE REVIEW

**Comment by jmshal** [comment]
Tested in Chrome, Safari and Firefox.

File: src/bugsnag.js [hunk 1]
```diff
@@ -66,6 +78,36 @@
   // ### Manual error notification (public methods)
   //
 
+  // #### Bugsnag.autoNotifyException
```
Comment (bengourley) [inline src/bugsnag.js]: this should say asyncNotifyException

File: src/bugsnag.js [hunk 2]
```diff
@@ -29,6 +29,18 @@
     // is configurable via the `maxDepth` setting.
     maxPayloadDepth = 5;
 
+  var stacktraceName;
+  try {
+    throw Error('');
+  } catch(err) {
+    stacktraceName = (
```
Comment (bengourley) [inline src/bugsnag.js]: Some comment on which browsers use which error property for the stack trace would be nice.

**Comment by jmshal** [comment]
Thanks @bengourley! I have my own doubts about this actually making it into the notifier. I feel at this point that it would be better if it was detailed within the docs (as a snippet), simply because of it's small size & very specific use-case. Although I do see this as a hack, I also feel like this is the best we're going to get, at least for a while - I spent a lot of time, and tried a bunch of different methods, but I settled on this as it was the least "involved". I usually like to stay optimistic about these things, but I don't think I've really seen this area of JS evolve much.

**Comment by bengourley** [comment]
> I feel at this point that it would be better if it was detailed within the docs (as a snippet)

I very much agree with this.

I'll close this PR but thanks your efforts in this area! Async tracing is definitely an area JS needs to catch up in… Node's [async hooks](https://nodejs.org/api/async_hooks.html) are a start but it sounds like TC39 are (or will be) [working on something](https://github.com/node-forward/discussions/issues/28#issue-53809642) for native JS, which means we're probably in for a rocky ride of incompatibility/competing standards in the future!

**Pull request closed without merging**